From 759492e8dd40fb6cdf355819df2ea30a44f3e072 Mon Sep 17 00:00:00 2001
From: Clayton Smith <argilo@gmail.com>
Date: Sun, 30 Oct 2022 11:53:04 -0400
Subject: [PATCH] cmake: Ensure that target names are unique

The current 20-bit hash is prone to collisions. Using the full hash
avoids that problem.

Signed-off-by: Clayton Smith <argilo@gmail.com>
---
 cmake/Modules/GrPython.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Modules/GrPython.cmake b/cmake/Modules/GrPython.cmake
index 9ee1a66f7..8f113273b 100644
--- a/cmake/Modules/GrPython.cmake
+++ b/cmake/Modules/GrPython.cmake
@@ -190,7 +190,7 @@ function(GR_UNIQUE_TARGET desc)
     file(RELATIVE_PATH reldir ${PROJECT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR})
     execute_process(
         COMMAND ${PYTHON_EXECUTABLE} -c "import re, hashlib
-unique = hashlib.md5(b'${reldir}${ARGN}').hexdigest()[:5]
+unique = hashlib.md5(b'${reldir}${ARGN}').hexdigest()
 print(re.sub('\\W', '_', r'${desc} ${reldir} ' + unique))"
         OUTPUT_VARIABLE _target
         OUTPUT_STRIP_TRAILING_WHITESPACE)
-- 
2.38.1

